src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -369,14 +459,23 @@ export class ObjectExplorerService implements IObjectExplorerService {
 						sessionId: session.sessionId,
 						nodes: [],
 						nodePath: nodePath,
-						errorMessage: undefined
+						errorMessage: undefined,
+						providerId: providerId
 					});
 				}
 			});
 		}
 
 		let provider = this._providers[providerId];
 		if (provider) {
+			let nodeProviders = this._nodeProviders[providerId];
+			if (nodeProviders) {
+				for (let nodeProvider of nodeProviders) {
+					nodeProvider.handleSessionClose({
```

Comment:
Double-checking this is now a notification, not anything that's awaitable?

Reply:
Looking at the API, it's awaitable so we need to at a minimum add a `.then(ok => undefined, error => // log error)`